BoostRunClub/Services/RunningService/RunningService.swift

```diff
@@ -0,0 +1,133 @@
+//
+//  RunningService.swift
+//  BoostRunClub
+//
+//  Created by 김신우 on 2020/12/20.
+//
+
+import Combine
+import Foundation
+
+class RunningService: RunningServiceType {
+    private(set) var cancellables = Set<AnyCancellable>()
+    private(set) var motionService: RunningMotionServiceable
+    private(set) var dashBoardService: RunningDashBoardServiceable
+    private(set) var recordService: RunningRecordServiceable
+
+    private var startTime: Date?
+
+    private var autoStateChangeable = true
+    private(set) var isStarted = false
+    let runningResultSubject = PassthroughSubject<(activity: Activity, detail: ActivityDetail)?, Never>()
+
+    let runningStateSubject = CurrentValueSubject<MotionType, Never>(.running)
+    var runningEventSubject = PassthroughSubject<RunningEvent, Never>()
+
+    var goalSubscription: AnyCancellable?
+
+    init(
+        motionProvider: RunningMotionServiceable,
+        dashBoard: RunningDashBoardServiceable,
+        recoder: RunningRecordServiceable
+    ) {
+        dashBoardService = dashBoard
+        recordService = recoder
+        motionService = motionProvider
+
+        dashBoard.runningStateSubject
+            .sink { [weak self] in
+                self?.recordService.addState($0)
+            }
+            .store(in: &cancellables)
+
+        motionProvider.motionTypeSubject
+            .receive(on: RunLoop.main)
+            .throttle(for: 2, scheduler: RunLoop.main, latest: true)
+            .filter { [weak self] _ in self?.autoStateChangeable ?? false }
+            .filter { [weak self] in $0 != self?.runningStateSubject.value }
+            .sink { [weak self] in
+                self?.runningStateSubject.send($0)
+                switch $0 {
+                case .running:
+                    dashBoard.setState(isRunning: true)
+                case .standing:
+                    dashBoard.setState(isRunning: false)
+                }
+            }
+            .store(in: &cancellables)
+    }
+
+    func start() {
+        recordService.clear()
+        dashBoardService.clear()
+
+        isStarted = true
+        startTime = Date()
+        autoStateChangeable = true
+
+        dashBoardService.start()
+        motionService.start()
+
+        runningEventSubject.send(.start)
+    }
+
+    func stop() {
+        let endTime = Date()
+
+        dashBoardService.stop()
+
+        isStarted = false
+
+        let activityInfo = recordService.save(startTime: startTime, endTime: endTime)
+        runningResultSubject.send(activityInfo)
+        runningEventSubject.send(.stop)
+    }
+
+    func pause(autoResume: Bool) {
+        autoStateChangeable = autoResume
+
+        dashBoardService.setState(isRunning: false)
+        runningEventSubject.send(.pause)
+    }
+
+    func resume() {
+        autoStateChangeable = true
+
+        dashBoardService.setState(isRunning: true)
+        runningEventSubject.send(.resume)
+    }
+
+    func setGoal(_ goalInfo: GoalInfo?) {
```

Comment:
```suggestion
    func setGoal(_ goalInfo: GoalInfo) {
```
Optional이 아니어도 컴파일 에러는 없네요.
가능하면 init 시점에 GoalInfo를 주입하는 것으로 리팩토링 해봐도 좋을 것 같아요.